db_stress_tool/expected_state.cc

```diff
@@ -67,20 +148,21 @@ int ExpectedState::DeleteRange(int cf, int64_t begin_key, int64_t end_key,
 }
 
 bool ExpectedState::Exists(int cf, int64_t key) {
-  // UNKNOWN_SENTINEL counts as exists. That assures a key for which overwrite
+  // Pending states count as "exist". That assures a key for which overwrite
   // is disallowed can't be accidentally added a second time, in which case
   // SingleDelete wouldn't be able to properly delete the key. It does allow
   // the case where a SingleDelete might be added which covers nothing, but
   // that's not a correctness issue.
   uint32_t expected_value = Value(cf, key).load();
-  return expected_value != SharedState::DELETION_SENTINEL;
+  return PendingWrite(expected_value) || PendingDelete(expected_value) ||
```

Comment:
`PendingWrite() || !IsDeleted()` feels like enough - is it not?

Reply:
Fixed